solver/src/solver/http_solver/model.rs

```diff
@@ -26,44 +28,56 @@ pub struct OrderModel {
 }
 
 #[derive(Debug, Serialize)]
-pub struct UniswapModel {
-    pub token1: String,
-    pub token2: String,
-    #[serde(with = "serde_with::rust::display_fromstr")]
-    pub balance1: u128,
-    #[serde(with = "serde_with::rust::display_fromstr")]
-    pub balance2: u128,
-    #[serde(with = "serde_with::rust::display_fromstr")]
-    pub fee: f64,
+pub struct PoolModel {
+    #[serde(flatten)]
+    pub parameters: PoolParameters,
     pub cost: CostModel,
     pub mandatory: bool,
 }
 
+#[derive(Debug, Serialize)]
+#[serde(tag = "pool_type")]
+pub enum PoolParameters {
+    ConstantProduct(ConstantProductPoolParameters),
+}
+
+#[derive(Debug, Serialize)]
+pub struct ConstantProductPoolParameters {
+    pub reserves: HashMap<H160, U256Wrapper>,
+    pub fee: BigDecimal,
```

Comment:
You mentioned today that this can be moved out to the common parameters in `PoolModel`.

Reply:
Yes, although I have done it in the following PR #775. The one that actually introduces a reason for moving it.